src/index.rs

```diff
@@ -1135,6 +1137,51 @@ impl Index {
         self.enum_qualified_variables.get(&qualified_name.to_lowercase())
     }
 
+    /// Tries to return an enum variant defined within a POU
+    pub fn find_enum_variant_in_pou(&self, pou: &str, variant: &str) -> Option<&VariableIndexEntry> {
+        self.get_enum_variants_in_pou(pou).into_iter().find(|it| it.name == variant)
+    }
+
+    /// Returns all enum variants defined within a POU
+    pub fn get_enum_variants_in_pou(&self, pou: &str) -> Vec<&VariableIndexEntry> {
+        self.enum_qualified_variables
+            .keys()
+            .filter(|key| key.starts_with(&internal_type_name(pou, "").to_lowercase()))
+            .flat_map(|key| self.find_fully_qualified_variable(key))
+            .collect()
+    }
+
+    /// Returns all enum variant values and their names for the given enum.
+    ///
+    /// For example `TYPE Color : (red, green, blue := 5); END_TYPE` would return `[(red, 0), (green, 1), (blue, 5)]`
+    /// when calling this method.
+    pub fn get_enum_variants(&self, variable: &VariableIndexEntry) -> Vec<(&str, &str, i128)> {
+        let mut values = Vec::new();
+        let qualified_name = variable.data_type_name.to_lowercase();
+
+        // Given `__main_color.red, ..., __main_color.blue`, we want ALL values starting with `__main_color`
+        let keys = self
+            .enum_qualified_variables
+            .keys()
+            .filter(|key| key.split('.').next().is_some())
+            .filter(|prefix| prefix.starts_with(&qualified_name))
+            .collect::<Vec<_>>();
```

Comment:
You probably did not need to collect here as the next step is a transformation of keys